components/script/dom/bindings/str.rs

```diff
@@ -134,30 +134,36 @@ pub fn extended_filtering(tag: &str, range: &str) -> bool {
             }
         }
 
+        let mut current_tag_subtag = tag_iter.next();
+
         // step 3
         for range_subtag in range_iter {
             // step 3a
             if range_subtag.eq_ignore_ascii_case("*") {
                 continue;
             }
-            loop {
-                match tag_iter.next() {
-                    Some(tag_subtag) => {
-                        // step 3c
-                        if range_subtag.eq_ignore_ascii_case(tag_subtag) {
-                            break;
+            match tag_iter.next() {
```

Comment:
I thought we agreed that this shoud match against `current_tag_subtag.clone()` instead?